qt.BUILD

```diff
@@ -1,90 +1,71 @@
-package(default_visibility = ["//visibility:public"])
+load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")
 
-cc_library(
-    name = "qt_core",
-    hdrs = glob(["QtCore/**"]),
-    includes = ["."],
-    linkopts = [
-        "-lQt5Core",
-    ],
-)
+QT_LIBRARIES = [
+    ("core", "QtCore", "Qt5Core", []),
+    ("network", "QtNetwork", "Qt5Network", []),
+    ("widgets", "QtWidgets", "Qt5Widgets", [":qt_core", ":qt_gui"]),
+    ("quick", "QtQuick", "Qt5Quick", [":qt_gui", ":qt_qml", ":qt_qml_models"]),
+    ("qml", "QtQml", "Qt5Qml", [":qt_core", ":qt_network"]),
+    ("qml_models", "QtQmlModels", "Qt5QmlModels", []),
+    ("gui", "QtGui", "Qt5Gui", [":qt_core"]),
+    ("opengl", "QtOpenGL", "Qt5OpenGL", []),
+]
 
-cc_library(
-    name = "qt_network",
-    hdrs = glob(["QtNetwork/**"]),
-    includes = ["."],
-    linkopts = [
-        "-lQt5Network",
-    ],
-)
+[
+    cc_library(
+        name = "qt_%s_linux" % name,
+        # When being on Windows this glob will be empty
+        hdrs = glob(["%s/**" % include_folder], allow_empty=True),
+        includes = ["."],
+        linkopts = ["-l%s" % library_name],
+        # Available from Bazel 4.0.0
+        # target_compatible_with = ["@platforms//os:linux"],
+    )
+    for name, include_folder, library_name, _ in QT_LIBRARIES
+]
 
-cc_library(
-    name = "qt_widgets",
-    hdrs = glob(["QtWidgets/**"]),
-    includes = ["."],
-    linkopts = [
-        "-lQt5Widgets",
-    ],
-    deps = [":qt_core", ":qt_gui"],
-)
+[
+    cc_import(
+        name = "qt_%s_windows_import" % name,
```

Comment:
I have very little knowledge of windows, so please forgive my ignorance. What is the use case for having separate _import and _windows targets if _windows just depends on the _import?

Reply:
I will have to revisit what was needed, I think was because of some limitation with cc_import and include files. However this was fix in Bazel, I would have to check in which version.
Here the question would be, what is the minimum version of Bazel that you would like to support? You might have seen that some parts of the code are commented out because it would require Bazel 4.0.

Reply:
I found it, this is the reason why we need to have cc_import and cc_library, basically because cc_impor does not suppor includes
https://github.com/bazelbuild/bazel/issues/4748

Reply:
Thanks for the link, that makes sense. Is it expected that the cc_import targets will ever be needed directly by users or is it more of an implementation detail? If the latter, I'd add a `_` prefix to the target name as a hint.

Reply:
It is more an implementation detail, it is the same for the ones ending with _windows and with _linux. That is the reason why they have the visibility as private (default). The only one that is public is the qt_library.
For me it is ok if you want to put the `_` prefix, I just did not do it because in the projects that I work I use it only for private macros and rules, when it comes to targets I'm use to just use the visibility flags.